mern-ecommerce/controllers/category.js

```diff
@@ -68,3 +68,36 @@ exports.list = (req, res) => {
     res.json(data);
   });
 };
+
+exports.addProductToCategory = (req, res) => {
+  const { categoryId } = req.params;
+  const { productId } = req.body;
+
+  Category.findById(categoryId, (err, category) => {
+    if (err || !category) {
```

Comment:
The error handling block is correctly checking for both the error and the absence of the category. However, it would be better to handle these two scenarios separately for more informative error messages. 

Instead of returning a generic "Category doesn't exist" error, consider returning a more specific error message based on whether the error is due to a database error or the category not being found. For example:

```
if (err) {
  return res.status(400).json({
    error: errorHandler(err),
  });
}
if (!category) {
  return res.status(404).json({
    error: "Category not found",
  });
}
```

This way, the client can receive more informative error messages and handle them accordingly.